services/src/main/java/org/keycloak/services/resources/admin/GroupResource.java

```diff
@@ -146,10 +146,6 @@ public void deleteGroup() {
     public Response addChild(GroupRepresentation rep) {
         this.auth.groups().requireManage(group);
 
-        if (group.getSubGroupsStream().map(GroupModel::getName).anyMatch(Predicate.isEqual(rep.getName()))) {
```

Comment:
This change needs to be accompanied with a change that interprets 409 correctly in admin UI

Reply:
I believe it should work the same. Look at the next line, there was:
```return ErrorResponse.exists("Parent already contains subgroup named '" + rep.getName() + "'");```

which should be the same 409, but I will check.